test/index.test.ts

```diff
@@ -316,3 +316,14 @@ test('task beforeAll, afterAll, beforeEach, afterEach', async () => {
   expect(afterEach.mock.calls.length).toBe(iterations * 2);
   expect(beforeEach.mock.calls.length).toBe(afterEach.mock.calls.length);
 });
+
+test('task with promiseLike return', async () => {
```

Comment:
we are lacking a test, where a sync function returns a promise

Reply:
This test actually does that

Reply:
Why do we then bother to cover various types if we dont test for them.

This is what I want as an additional test (not tested!):

```js
test('sync task which returns a promise', async () => {
  const bench = new Bench();

  bench.add('foo', () => {
    return new Promise(resolve => setTimeout(resolve, 100))
  });

  await bench.run();
  expect(bench.getTask('foo')!.result!.mean).toBeGreaterThan(100);
});

Reply:
Okay, I'll add, it's the same for me

```js
bench.add('foo', () => ({
    then: async (resolve: () => void) => setTimeout(resolve, 100),
  }));
```

Reply:
still, add that case please.